ostd/src/arch/x86/cpu.rs

```diff
@@ -372,103 +344,107 @@ impl UserContextApiInternal for UserContext {
-            pub const $name : CpuException = CpuException{
-                number: $exception_num,
-                typ: CpuExceptionType::$exception_type,
-            };
-        )*
+            /// Get the exception from the exception number.
+            pub fn from_num(num: u16) -> Self {
+                match num {
+                    $(
+                        $exception_num => Self::$name,
+                    )*
+                    n => CpuException::NotExplicitInISA(n),
+                }
+            }
+        }
     }
 }
 
-// We also defined the RESERVED Exception so that we can easily use the index of EXCEPTION_LIST to get the Exception
 define_cpu_exception!(
-    [DIVIDE_BY_ZERO = 0, Fault],
-    [DEBUG = 1, FaultOrTrap],
-    [NON_MASKABLE_INTERRUPT = 2, Interrupt],
-    [BREAKPOINT = 3, Trap],
-    [OVERFLOW = 4, Trap],
-    [BOUND_RANGE_EXCEEDED = 5, Fault],
-    [INVALID_OPCODE = 6, Fault],
-    [DEVICE_NOT_AVAILABLE = 7, Fault],
-    [DOUBLE_FAULT = 8, Abort],
-    [COPROCESSOR_SEGMENT_OVERRUN = 9, Fault],
-    [INVAILD_TSS = 10, Fault],
-    [SEGMENT_NOT_PRESENT = 11, Fault],
-    [STACK_SEGMENT_FAULT = 12, Fault],
-    [GENERAL_PROTECTION_FAULT = 13, Fault],
-    [PAGE_FAULT = 14, Fault],
-    [RESERVED_15 = 15, Reserved],
-    [X87_FLOATING_POINT_EXCEPTION = 16, Fault],
-    [ALIGNMENT_CHECK = 17, Fault],
-    [MACHINE_CHECK = 18, Abort],
-    [SIMD_FLOATING_POINT_EXCEPTION = 19, Fault],
-    [VIRTUALIZATION_EXCEPTION = 20, Fault],
-    [CONTROL_PROTECTION_EXCEPTION = 21, Fault],
-    [RESERVED_22 = 22, Reserved],
-    [RESERVED_23 = 23, Reserved],
-    [RESERVED_24 = 24, Reserved],
-    [RESERVED_25 = 25, Reserved],
-    [RESERVED_26 = 26, Reserved],
-    [RESERVED_27 = 27, Reserved],
-    [HYPERVISOR_INJECTION_EXCEPTION = 28, Fault],
-    [VMM_COMMUNICATION_EXCEPTION = 29, Fault],
-    [SECURITY_EXCEPTION = 30, Fault],
-    [RESERVED_31 = 31, Reserved]
+    // Hardware exceptions
+    [DivideByZero = 0, Fault],
+    [Debug = 1, FaultOrTrap],
+    [NonMaskableInterrupt = 2, Interrupt],
+    [Breakpoint = 3, Trap],
+    [Overflow = 4, Trap],
+    [BoundRangeExceeded = 5, Fault],
+    [InvalidOpcode = 6, Fault],
+    [DeviceNotAvailable = 7, Fault],
+    [DoubleFault = 8, Abort],
+    [CoprocessorSegmentOverrun = 9, Fault], // Not supported on AMD64
+    [InvalidTss = 10, Fault],
+    [SegmentNotPresent = 11, Fault],
+    [StackSegmentFault = 12, Fault],
+    [GeneralProtectionFault = 13, Fault],
+    [PageFault = 14, Fault],
+    [X87FloatingPointException = 16, Fault],
+    [AlignmentCheck = 17, Fault],
+    [MachineCheck = 18, Abort],
+    [SimdFloatingPointException = 19, Fault],
+    [VirtualizationException = 20, Fault],
+    [ControlProtectionException = 21, Fault],
+    [HypervisorInjectionException = 28, Fault],
+    [VmmCommunicationException = 29, Fault],
+    [SecurityException = 30, Fault],
+    // User-interpreted interrupts
+    [Syscall = 256, Interrupt]
```

Comment:
I don't think treating syscall and Non-maskable Interrupt as the same type is a good idea. Maybe we can create a new type for the syscall.

Reply:
I hesitated when implementing it either. I suppose that making it standalone is better